src/main/java/io/jenkins/plugins/junit/checks/JUnitChecksPublisher.java

```diff
@@ -0,0 +1,148 @@
+import io.jenkins.plugins.checks.api.ChecksPublisherFactory;
+import io.jenkins.plugins.checks.api.ChecksStatus;
+import io.jenkins.plugins.util.JenkinsFacade;
+import java.util.List;
+import jnr.ffi.Struct;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+@Restricted(NoExternalUse.class)
+public class JUnitChecksPublisher {
+    public static final String SEPARATOR = ", ";
+    private final TestResultAction action;
+    private final JenkinsFacade jenkinsFacade;
+    private final TestResultSummary summary;
+
+    public JUnitChecksPublisher(final TestResultAction action, final TestResultSummary summary) {
+        this(action, summary, new JenkinsFacade());
+    }
+
+    @VisibleForTesting
+    JUnitChecksPublisher(final TestResultAction action, final TestResultSummary summary, final JenkinsFacade jenkinsFacade) {
+        this.jenkinsFacade = jenkinsFacade;
+        this.action = action;
+        this.summary = summary;
+    }
+
+    public void publishChecks(TaskListener listener) {
+        ChecksPublisher publisher = ChecksPublisherFactory.fromRun(action.run, listener);
+        publisher.publish(extractChecksDetails());
+    }
+
+    @VisibleForTesting
+    ChecksDetails extractChecksDetails() {
+        String testsURL = DisplayURLProvider.get().getTestsURL(action.run);
+        String text = extractChecksText(testsURL);
+        ChecksOutput output = new ChecksOutput.ChecksOutputBuilder()
+                .withTitle(extractChecksTitle())
+                .withSummary("<sub>Send us [feedback](https://github.com/jenkinsci/junit-plugin/issues)")
+                .withText(text)
+                .build();
+
+        return new ChecksDetails.ChecksDetailsBuilder()
+                .withName("Tests")
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(summary.getFailCount() > 0 ? ChecksConclusion.FAILURE : ChecksConclusion.SUCCESS)
+                .withDetailsURL(testsURL)
+                .withOutput(output)
+                .build();
+    }
+
+    private String extractChecksText(String testsURL) {
+        StringBuilder builder = new StringBuilder();
+        if (summary.getFailCount() > 0) {
+            List<CaseResult> failedTests = action.getResult().getFailedTests();
+            
+            if (failedTests.size() > 10) {
+                failedTests = failedTests.subList(0, 9);
+            }
+            
+            failedTests.forEach(failedTest -> mapFailedTestToTestReport(builder, failedTest));
+            if (summary.getFailCount() > 10) {
+                builder.append("\n")
+                        .append(summary.getFailCount() - 10)
+                        .append(" more test results are not shown here, view them on [Jenkins](")
+                        .append(testsURL).append(")");
```

Comment:
Does this `testsURL` links to the JUnit summary page? If so, I think we should use different links for here and the `detailsURL`, maybe just the build page links for the `detailsURL`?

Reply:
It links to the same place, I don’t think there’s any need to link differently the only place one would go it the test report from this check